src/diagnostic-channel-publishers/src/pg.pub.ts

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+import {channel, IModulePatcher, PatchFunction} from "diagnostic-channel";
+import {EventEmitter} from "events";
+
+// copy the pg.Result type: https://node-postgres.com/api/result
+export interface IPostgresResult {
+    rowCount: number;
+    command: string;
+}
+
+export interface IPostgresData {
+    query: {
+        text?: string;
+        plan?: string;
+        preparable?: {
+            text: string;
+            args: any[];
+        }
+    };
+    database: {
+        host: string;
+        port: string;
+    };
+    result?: IPostgresResult;
+    duration: number;
+    error?: Error;
+}
+
+type PostgresCallback = (err: Error, res: IPostgresResult) => any;
+
+function postgres6PatchFunction(originalPg, originalPgPath) {
+    const originalClientQuery = originalPg.Client.prototype.query;
+
+    // this keeps track of user-supplied callback => patched callbacks so we don't keep
+    // re-patching the same callback over and over for library functions
+    // IMPORTANT: use a WeakMap so that we don't keep references to temporary/otherwise unreferenced functions
+    const patchedCallbacks = new WeakMap<PostgresCallback, PostgresCallback>();
```

Comment:
so, is WeakMap going to work on down-level node versions?  Better approach may be to tag the function with the original, e.g., something like this:

```
function stamp(cb, wrapper) {
   cb.__diagnosticChannelProxy = wrapper;
   wrapper.__diagnosticChannelOriginal = cb;
}
```
 
Then you can easily detect if any given function has been proxied, and you can easily traverse between proxy & proxied function.  And it works across node versions with no concerns about memory leaks. 

Also, should probably be consistent w/ this approach in all the functions we proxy.